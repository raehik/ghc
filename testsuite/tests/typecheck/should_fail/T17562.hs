{-# LANGUAGE QuantifiedConstraints, MultiParamTypeClasses #-}

module T17562 where

class (forall a. a b ~ a c) => C b c
