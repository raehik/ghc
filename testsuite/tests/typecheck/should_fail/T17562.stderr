
T17562.hs:5:1: error:
    • Uninferrable type variable ‘k0’ in
        forall (a :: * -> k0). (a b :: k0) ~ (a c :: k0)
    • In the class declaration for ‘C’
